zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -252,136 +245,36 @@ pub async fn run() -> Result<()> {
-        "performing full sync...",
-    );
-
-    let full_sync_path =
-        copy_state_and_perform_full_sync(network, zebrad_state_path.as_ref()).await?;
-
-    tracing::info!(?full_sync_path, "loading transactions...");
-
-    let transactions =
-        load_transactions_from_block_after(partial_sync_height, network, full_sync_path.as_ref())
-            .await?;
-
-    Ok(transactions)
-}
-
-/// Loads transactions from a block that's after the specified `height`.
+/// Loads transactions from 1-3 block(s) after the chain tip of the cached state.
 ///
-/// Starts at the block after the block at the specified `height`, and stops when it finds a block
-/// from where it can load at least one non-coinbase transaction.
+/// Returns a list of valid transactions that are not in any of the blocks finalized to disk
+/// in the original `zebrad_state_path`.
 ///
-/// # Panics
+/// ## Panics
 ///
-/// If the specified `zebrad_state_path` contains a chain state that's not synchronized to a tip that's
-/// after `height`.
+/// If the provided `test_type` doesn't need an rpc server and cached state
 #[tracing::instrument]
-async fn load_transactions_from_block_after(
-    height: block::Height,
+async fn load_transactions_from_future_blocks(
     network: Network,
-    zebrad_state_path: &Path,
+    test_type: TestType,
+    test_name: &str,
 ) -> Result<Vec<Arc<Transaction>>> {
-    let (_read_write_state_service, mut state, latest_chain_tip, _chain_tip_change) =
-        start_state_service_with_cache_dir(network, zebrad_state_path).await?;
-
-    let tip_height = latest_chain_tip
-        .best_tip_height()
-        .ok_or_else(|| eyre!("State directory doesn't have a chain tip block"))?;
-
-    assert!(
-        tip_height > height,
-        "Chain not synchronized to a block after the specified height",
-    );
-
-    let mut target_height = height.0;
-    let mut transactions = Vec::new();
-
-    while transactions.len() < max_sent_transactions() {
-        let new_transactions =
-            load_transactions_from_block(block::Height(target_height), &mut state).await?;
-
-        if let Some(mut new_transactions) = new_transactions {
-            new_transactions.retain(|transaction| !transaction.is_coinbase());
-            transactions.append(&mut new_transactions);
-        } else {
-            tracing::info!(
-                "Reached the end of the finalized chain\n\
-                 collected {} transactions from {} blocks before {target_height:?}",
-                transactions.len(),
-                target_height - height.0 - 1,
-            );
-            break;
-        }
-
-        target_height += 1;
-    }
+    let transactions = get_future_blocks(network, test_type, test_name, 3)
+        .await?
+        .into_iter()
+        .map(|block| block.transactions)
+        .reduce(|mut p, mut n| {
+            if (p.len() + n.len()) < max_sent_transactions() {
+                p.append(&mut n)
+            };
```

Comment:
This doesn't quite do what the previous test did:
1. Coinbase transactions are invalid in the mempool and must be skipped
2. If there are 10 transactions in the first block, the returned list will be empty. Instead, we want to get as many transactions as are available, then limit the list to 10 transactions.

```suggestion
        // Skip coinbase transactions
        .flat_map(|block| block.transactions[1..])
        .take(max_sent_transactions())
        .cloned()
        .collect();
```

Reply:
I missed filtering out the coinbase transaction, I like this better, thank you. I thought transactions may have unordered dependencies on other transactions in that block, but looking at the transactions' verifier, that is not the case. 

Applied a slightly modified version in [`d72ea40`](https://github.com/ZcashFoundation/zebra/pull/5589/commits/d72ea40650530d1dfa5e3bdd73d9d146eb28fba8) and [`c34a955`](https://github.com/ZcashFoundation/zebra/pull/5589/commits/c34a95506f00f3830e3a299c3ef35862ee90d169)

Note: while `fold` would return an empty list, `reduce` uses the first element as the accumulator and calls the provided fn starting from the second element.

Reply:
> I thought transactions may have unordered dependencies on other transactions in that block, but looking at the transactions' verifier, that is not the case.

Later transactions in a block can spend the transparent outputs of earlier transactions. That is checked in the state, where we are no longer verifying transactions in parallel.